### PR TITLE
Adds playtime requirements for antag roles

### PR DESCRIPTION
**Pull request by Kyep**
🆑 Kyep
rscadd: Antag roles now have playtime requirements. 
/🆑

- All antag roles now have playtime requirements.
- Solo antags require 3 hours, partner antags (e.g: guardian) require 5h, team-based antags (shadowlings, nukeops, abductors, ERT, etc) require 10h. 
- If someone can't choose a certain role, it will display as "[IN X HRS]" in their preferences panel.
- These restrictions are enabled by the same config as job playtime requirements are. Once this PR is merged, we can remove USE_AGE_RESTRICTION_FOR_JOBS from our config.

This PR should dramatically reduce the number of clueless newbies playing Emergency Response Team officers, Nuclear Operatives, etc.

### CODE REVIEW

File: code/game/jobs/job_exp.dm [hunk 1]
```diff
@@ -1,3 +1,40 @@
...
+	ROLE_VAMPIRE = 3,
+	ROLE_BLOB = 3,
+	ROLE_REVENANT = 3,
+	ROLE_BORER = 3,
+	ROLE_NINJA = 3,
+	ROLE_MORPH = 3,
+
+	// DUO ANTAGS
+	ROLE_GUARDIAN = 5,
+	ROLE_DEMON = 5,
```
Comment (Fox-McCloud) [inline code/game/jobs/job_exp.dm]: Demons are not duo antags, at all.

File: code/game/jobs/job_exp.dm [hunk 2]
```diff
@@ -1,3 +1,40 @@
...
+
+	// TEAM ANTAGS
+	// Higher numbers here, because they require more experience to be played correctly
+	ROLE_SHADOWLING = 10,
+	ROLE_REV = 10,
+	ROLE_OPERATIVE = 10,
+	ROLE_CULTIST = 10,
+	ROLE_RAIDER = 10,
+	ROLE_ALIEN = 10,
+	ROLE_ABDUCTOR = 20,
```
Comment (Fox-McCloud) [inline code/game/jobs/job_exp.dm]: Abductors should be 10 if the others are 10

File: code/game/jobs/job_exp.dm [hunk 3]
```diff
@@ -1,3 +1,40 @@
+// Playtime requirements for special roles (hours)
+
+var/global/list/role_playtime_requirements = list(
+	// CREW-FRIENDLY ROLES
+	ROLE_PAI = 0,
+	ROLE_POSIBRAIN = 5, // Same as cyborg job.
+	ROLE_SENTIENT = 10,
```
Comment (Fox-McCloud) [inline code/game/jobs/job_exp.dm]: That's a lot of playtime for something incredibly simple

File: code/game/jobs/job_exp.dm [hunk 4]
```diff
@@ -1,3 +1,40 @@
...
+	ROLE_ERT = 20, // High, because they're team-based, and we want ERT to be robust
```
Comment (Fox-McCloud) [inline code/game/jobs/job_exp.dm]: High hopes man, high hopes.

**Comment by Aracino** [comment]
Will this only affect new players? Since when i got back after a break, it still needed me to re-get hoirs for certaon roles

**Comment by Kyep** [comment]
All existing playtime will count for this. If you currently have head roles unlocked, you'll have all of the antag roles unlocked, too.

**Comment by Twinmold93** [comment]
Looks good to me.

**Comment by granodd** [comment]
Does the time requirements listed here actually translate into a long enough period before someone is "experienced" enough to play those roles? I say this because the usual shift is ten hours, which if you played one shift every day on the weekdays, could be done in a week. I didn't discover how to do many of the jobs (horrible example I know, as some learn faster/slower than others) until after a good couple of weeks into playing this game. I know we don't want to make the required time too long/ridiculous, but I'm just concerned if this would solve the issue of greenhorns getting placed into the roles.

**Comment by Taaketa** [comment]
The previous job time adjustment was set with flexibility, this will be no different.

**Comment by Anoonki** [comment]
[Woah, didn't know we were FTL13, Granodd.](http://i.imgur.com/1eNwm17.png)

**Comment by Kyep** [comment]
@granodd Before this PR is merged, all you have to do to unlock everything is connect to the server once, then come back 30 days later, and presto, every antag role will be unlocked. Without you having played the game for a single minute. 
Under this PR, you'd have to actually put in a few hours on the server to unlock the antag roles. This PR's system is a **massive** improvement over what we have now. Could the number requirements be higher? Sure. But the numbers are intentionally set rather low, for the moment. We can always tweak them later if we want. Right now, the purpose of this is just to get away from the god-awful "days since registration" system we have now.

File: code/game/jobs/job_exp.dm (hunk 2 above)
Comment (SamHPurp) [inline code/game/jobs/job_exp.dm]: @Fox-McCloud Currently Abductors are 30 days, Cultists are 21, aliens are 21 etc.

**Comment by KingPhilipIII** [comment]
Should add a new antag role as nuclear leader. Requires several hours of playtime as a nukie. I know you can just hand the ID off to another person but why not just nip it quickly. Not everyone is willing to comply either. "NO FUCK YOU I'M THE LEADER NOW LETS DO OUR STUPID PLAN WHILE I CANT FIGURE OUT THE TC SYSTEM

**Comment by SamHPurp** [comment]
@KingPhilipIII - I agree, but maybe out of scope of this, which just changes the requirements for antags to time played... a followup PR splitting out Leader makes sense to me.

**Comment by KingPhilipIII** [comment]
@SamHPurp True, but I lack the ability to code myself, so I gotta mention it in relevant places and hope people notice.

**Comment by Fruerlund** [comment]
Seems good to me!

**Comment by Kyep** [comment]
@KingPhilipIII Not doing that yet because:
A) It would significantly complicate the nuclear team selection process.
B) It isn't a catch-all fix. Problematic team members are more common than problematic team leaders, so, for the moment, I address both by adding a minimum requirement that applies to team members, including team leaders.
If, after this PR is merged, I still see nuke op leaders being terrible, AND when I look, I see that they have, say, 10-20 hours of playtime, then I will consider upping the requirement for them to 20h. That, however, is not something I want to do before this PR is merged, and I've had a chance to see if this is enough to fix the problem.

**Comment by mrturkeytoe** [comment]
My own two cents, love it. Can we have the team leader canidacy selection seperated from the team member one though?

**Comment by Kyep** [comment]
@mrturkeytoe Same reply as I gave KingPhilipIII.

**Comment by Kyep** [comment]
@Fox-McCloud @Crazylemon64 Is this waiting on anything? This is another PR that has been sitting as-is, with no outstanding issues, for a month.

**Comment by Fox-McCloud** [comment]
It looks ok to me. @Crazylemon64

**Pull request merged**